### PR TITLE
[ONNX] replace AT_ASSERT with TORCH_INTERTNAL_ASSERT take 2

### LINKED ISSUE #84132: [ONNX] replace AT_ASSERT with TORCH_INTERNAL_ASSERT
Found the issue in pass/onnx folder.

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86405

Address the AT_ASSERT in torch/jit/csrc/serialization (ONNX related).

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
Thanks!

File: torch/csrc/jit/passes/onnx/pattern_conversion/pattern_conversion.cpp [hunk 1]
```diff
@@ -146,7 +146,7 @@ std::unordered_map<int64_t, ConvertedIndex> MergeSliceAndSelectToIndices(
           std::forward_as_tuple(index_tensor, aten::select));
       dim_offset++;
     } else {
-      AT_ERROR(
+      TORCH_INTERNAL_ASSERT(
```
Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/pattern_conversion/pattern_conversion.cpp:149]:
```suggestion
      TORCH_CHECK(false,
```

Use `TORCH_CHECK` for user input error. Also note the first argument of both `TORCH_CHECK` and `TORCH_INTERNAL_ASSERT` is for condition. `AT_ERROR` on the other hand only takes message string as argument.

**Review by BowenBao — APPROVED** [review]
🚀  with inline comment.

File: torch/csrc/jit/passes/onnx/pattern_conversion/pattern_conversion.cpp (hunk 1 above)
Comment (AllenTiTaiWang) [reply torch/csrc/jit/passes/onnx/pattern_conversion/pattern_conversion.cpp]: Done

**Pull request closed without merging**